### DISCUSSION TITLE
CPU benchmark

### DISCUSSION THREAD

**Discussion by dhpark0755** [Heerozh/spectre#10] [Q&A]
Hi, 
It seems the Spectre(CPU) doesn't outperform zipline in magnitude. I wonder if you have insight why this is the case? Since Spectre is using vectorized calculation while Zipline's computation is step-by-step, I would expect larger improvement from there.
Thanks,
DH

**Comment by Heerozh** [comment]
Because although zipline is split data into step by step, but only by date, doesn't impact performance much. 
Each step contains all stocks and historical data, and still can perform vectorized calculations.

**Comment by dhpark0755** [comment]
Thank you.

**Comment by quant2008** [comment]
如果使用分钟bar数据，那么数据量及其巨大，spectre一次性装进几千只股票的所有分钟数据会不会有内存问题？
zipline pipeline在这种场景下，是不是不会一次性全部装入，而是分步装入内存，内存消耗会少一些？ @Heerozh
